install.sh

```diff
@@ -62,134 +66,114 @@ function ask() {
 		prompt="${GREEN}y${NC}/${RED}n${NC}"
 	fi
 
-	while true; do
-		# Ask the question (not using "read -p" as it uses stderr not stdout)
-		echo -ne "$1 [$prompt] "
-
-		# Read the answer (use /dev/tty in case stdin is redirected from somewhere else)
-		if [[ -z "$DISABLE_PROMPTS" ]]; then
-			read -r reply < /dev/tty
-		else
-			echo "$default"
-			reply=$default
-		fi
-
-		# Default?
-		if [[ -z $reply ]]; then
-			reply=$default
-		fi
+	# Ask the question (not using "read -p" as it uses stderr not stdout)
+	echo -ne "$1 [$prompt] "
 
+	# Read the answer (use /dev/tty in case stdin is redirected from somewhere else)
+	if [[ -z "$DISABLE_PROMPTS" ]]; then
+		read -r reply < /dev/tty
+	else
+		echo "$default"
+		reply=$default
+	fi
+
+	# Default?
+	if [[ -z $reply ]]; then
+		reply=$default
+	fi
+
+	while true; do
 		# Check if the reply is valid
 		case "$reply" in
-			Y*|y*) return 0 ;;
-			N*|n*) return 1 ;;
+			Y*|y*)
+				answer=1
+				return 0	#return code for backwards compatibility
+				break
+			;;
+			N*|n*)
+				answer=0
+				return 1	#return code
+				break
+			;;
+			*)
+				echo -ne "$1 [$prompt] "
+				read -r reply < /dev/tty
+			;;
 		esac
 	done
 }
 
-function banner() {
-	echo -e "|------------------------|"
-	echo -e "|---${GREEN}Pacstall Installer${NC}---|"
-	echo -e "|------------------------|"
-	echo " "
-}
-
 if ! command -v apt &> /dev/null; then
 	fancy_message error "apt could not be found"
 	exit 1
 fi
 # Install wget and sudo (probably already installed but this is important for the tester)
 apt-get install -y -qq sudo wget
 
-# Colors
-RED='\033[0;31m'
-NC='\033[0m'
-GREEN='\033[0;32m'
-YELLOW='\033[1;33m'
-BLUE='\033[0;34m'
+echo -e "|------------------------|"
+echo -e "|---${GREEN}Pacstall Installer${NC}---|"
+echo -e "|------------------------|"
 
-banner
-
-printf "Checking for internet access: "
-
-wget -q --tries=10 --timeout=20 --spider https://github.com &
-PID=$!
-i=1
-sp="/-\|"
-printf ' '
-while [ -d /proc/$PID ]
-do
-	sleep 0.1
-	printf "\b${sp:i++%${#sp}:1}"
-done
-if [[ $? -eq 1 ]] ; then
+if ! nm-online -qx; then
```

Comment:
What if someone don't has `nm-online`? Some use `systemd-networkd` and some also use `connman`

Reply:
`NetworkManager` (which is the main app) ships with every linux distribution. It is independent of `systemd-networkd` ( They are totally different apps, and can work when either of them is not present in the system ). So it should work in every distribution.

Reply:
Well `Clear Linux` and many more use `systemd-networkd` and don't ship with `networkmanager`. And some people may remove `networkmanager` from their system and use `systemd-networkd` so you can set `wget` one as fallback